module/zfs/vdev_indirect.c

```diff
@@ -1447,6 +1451,14 @@ vdev_indirect_repair(zio_t *zio)
 			    ZIO_FLAG_IO_REPAIR | ZIO_FLAG_SELF_HEAL,
 			    NULL, NULL));
 
+			/*
+			 * If ic_error is set the current child does not have
+			 * a copy of the data, so suppress incrementing the
+			 * checksum counter.
+			 */
+			if (ic->ic_error == ESTALE)
+				return;
```

Comment:
```suggestion
				continue;
```

You don't want to return here since there may be splits on other vdevs to repair.